lab_2_tokenize_by_bpe/main.py

```diff
@@ -26,6 +37,18 @@ def collect_frequencies(
     :param end_of_word: a token that signifies the end of word
     :return: dictionary in the form of <preprocessed word: number of occurrences>
     """
+    if not isinstance(text, str) or\
+            not isinstance(start_of_word, str | None) or\
+            not isinstance(end_of_word, str):
+        return None
+    tokens_dictionary = {}
+    for token in text.split():
+        split_word = prepare_word(token, start_of_word, end_of_word)
+        if split_word is None:
+            return None
+        if split_word not in tokens_dictionary:
+            tokens_dictionary[split_word] = text.split().count(token)
```

Comment:
Consider doing split just once and saving result in a variable

Reply:
Done